src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -86,6 +87,16 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         this.disposables.push(this._onNotebookControllerSelected);
         this.isLocalLaunch = isLocalLaunch(this.configuration);
     }
+    public async getInteractiveController(): Promise<VSCodeNotebookController | undefined> {
+        // Fetch the active interpreter and use the matching controller
+        const api = await this.pythonApi.getApi();
+        const activeInterpreter = await api.getActiveInterpreter();
+
+        if (!activeInterpreter) {
+            return;
+        }
+        this.getOrCreateController(activeInterpreter, InteractiveWindowView)?.id;
```

Comment:
```suggestion
        return this.getOrCreateController(activeInterpreter, InteractiveWindowView);
```

Reply:
We need this change, otherwise the controller will never be returned from `getInteractiveController`.